indico/modules/events/surveys/fields/choices.py

```diff
@@ -136,3 +141,9 @@ def validators(self):
     def wtf_field_kwargs(self):
         return {'choices': [(x['id'], x['option']) for x in self.question.field_data['options']],
                 'coerce': lambda x: x}
+
+    def get_actual_answer(self, answer):
+        question_options = {option_dict['id']: option_dict['option']
+                            for option_dict in self.question.field_data['options']}
+        return [question_options[value] for value in answer if value in question_options]
```

Comment:
i think this will fail when you have an answer referencing an option that was deleted afterwards